### PR TITLE
Add onLeave callback to DragTarget

### PR DESCRIPTION
**Pull request by tlarsendataguy**
I am trying to implement a ListView with items that can be reordered by dragging and dropping.  I can respond to 'drag enter' events to adjust the UI by providing an OnWillAccept callback, but I currently have no way to easily know when a 'drag leave' event occurs.  Having this facility would allow the DragTarget to maintain its state, by itself, throughout the drag lifecycle and simplify the build process.

### CODE REVIEW

**Comment by Hixie** [comment]
This will need tests to make sure we don't accidentally revert this change at some point (make sure anything you care about is tested!).

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 1]
```diff
@@ -388,6 +394,10 @@ class DragTarget<T> extends StatefulWidget {
   /// Called when an acceptable piece of data was dropped over this drag target.
   final DragTargetAccept<T> onAccept;
 
+  /// Called when a given piece of data being dragged over this target leaves
+  /// the target.
+  final DragTargetLeave<T> onLeave;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/drag_target.dart]: We should probably add to the onWillAccept docs that that is called when a piece of data enters the target, and that that will be followed by either onAccept or onLeave.

**Comment by Hixie** [comment]
This patch looks great. Just some tests and then it should be good to land!
Thanks for your contribution!

**Comment by tlarsen7572** [comment]
Thanks for the feedback!  I have been having trouble getting the tests running on my local Flutter copy, so I got a bit lazy with this PR.  I will get them up and running and add some for this behavior.

**Comment by Hixie** [comment]
Let me know if you need any help getting tests to run. Should just be a matter of running `flutter test` in the `packages/flutter` directory.

**Comment by tlarsen7572** [comment]
I think I got my situation figured out.  I added a test for onLeave and took a stab at expanding the onWillAccept documentation.

**Comment by tlarsen7572** [comment]
By the way, neat testing framework for widgets.  I have always struggled with UI/Widget tests and don't do TDD on them as a result.  This was a good intro for me.

**Comment by Hixie** [comment]
Thanks for your contribution! LGTM

**Pull request merged**